02-KernelModules_CharacterDevice/chrdev.c

```diff
@@ -0,0 +1,232 @@
+#include <linux/kernel.h>
+#include <linux/proc_fs.h>
+#include <linux/sched.h>
+#include <asm/uaccess.h>
+#include <linux/slab.h>
+#include <linux/moduleparam.h>
+#include <linux/device.h>
+#include <linux/module.h>
+#include <linux/sysfs.h>
+
+#ifndef MODULE_NAME
+#define MODULE_NAME 	"chrdev"
+#endif
+
+#define BUFFER_SIZE	    1024 
+#define DEVICE_NAME 	"charDevice"
```

Comment:
```suggestion
#define DEVICE_NAME   "charDevice"
#define CLASS_NAME    "charClass"
```
Please use kernel code style https://www.kernel.org/doc/html/v4.10/process/coding-style.html 

Please use Check kernel code with checkpatch